test/e2e-olm/utils.go

```diff
@@ -0,0 +1,252 @@
+package e2e
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/constants"
+	"github.com/openshift/elasticsearch-operator/test/utils"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	retryInterval        = time.Second * 1
+	timeout              = time.Second * 300
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	elasticsearchCRName  = "elasticsearch"
+	kibanaCRName         = "kibana"
+)
+
+func registerSchemes(t *testing.T) {
```

Comment:
objs := []runtime.Object{
		&elasticsearch.ElasticsearchList{},
		&elasticsearch.KibanaList{},
		&consolev1.ConsoleLinkList{},
	}
	for _, obj := range objs {
		err := framework.AddToFrameworkScheme(elasticsearch.SchemeBuilder.AddToScheme, obj)
		if err != nil {
			log.Fatalf("failed to add custom resource scheme to framework: %v", err)
		}
	}

Reply:
Actually, this is a nice approach, but for some deep-in-k8s/client-go reason you cannot build the dynamic client if you use the runtime object interface.

Reply:
What's the simplest way for me to see it blow up?

Reply:
Just use your code snippet and run the e2e tests.

Reply:
In the first version, I missed the fact that the AddToScheme func also changes in the loop, so here you go:
```
	for _, arg := range []struct {
		addToSchemeFunc func(*runtime.Scheme) error
		obj             runtime.Object
	}{
		{
			elasticsearch.SchemeBuilder.AddToScheme,
			&elasticsearch.ElasticsearchList{},
		},
		{
			elasticsearch.SchemeBuilder.AddToScheme,
			&elasticsearch.ElasticsearchList{},
		},
		{
			consolev1.Install,
			&consolev1.ConsoleLinkList{},
		},
	} {
		err := framework.AddToFrameworkScheme(arg.addToSchemeFunc, arg.obj)
		if err != nil {
			t.Fatalf("failed to add custom resource scheme to framework: %v", err)
		}
	}
```